src/requestSchemas.ts

```diff
@@ -0,0 +1,210 @@
+      message: 'Must provide identifying information via either id, url, or identifier parameters'
+    });
+  }
+}
+
+const stringToBool = z
+  .union([z.enum(['true', 'false']), z.boolean()])
+  .transform(x => (typeof x === 'boolean' ? x : x === 'true'));
+const stringToNumber = z.coerce.number();
+const checkDate = (paramName: string) => {
+  return z.string().regex(DATE_REGEX, `${paramName} parameter is not a valid FHIR date`);
+};
+
+export const IdentifyingParameters = z
+  .object({
+    id: z.string(),
+    url: z.string().url(),
+    version: z.string(),
+    identifier: z.string()
+  })
+  .partial();
+
+export const PackageArgs = IdentifyingParameters.extend({
+  capability: z.string(),
+  'check-system-version': z.string(),
+  count: stringToNumber,
+  'force-system-version': z.string(),
+  'include-components': stringToBool,
+  'include-dependencies': stringToBool,
+  'include-terminology': stringToBool,
+  manifest: z.string(),
+  offset: stringToNumber,
+  'system-version': z.string()
+})
+  .partial()
+  .strict()
+  .superRefine(catchInvalidParams([catchMissingIdentifyingInfo], UNSUPPORTED_PACKAGE_ARGS));
+
+export const CommonDataRequirementsArgs = IdentifyingParameters.extend({
+  'check-system-version': z.string(),
+  'force-system-version': z.string(),
+  'include-components': stringToBool,
+  'include-dependencies': stringToBool,
+  manifest: z.string(),
+  parameters: z.string(),
+  periodEnd: checkDate('periodEnd'),
+  periodStart: checkDate('periodStart'),
+  'system-version': z.string()
+})
+  .partial()
+  .strict();
+
+export const MeasureDataRequirementsArgs = CommonDataRequirementsArgs.superRefine(
+  catchInvalidParams([catchMissingIdentifyingInfo], UNSUPPORTED_DATA_REQ_ARGS)
+);
+
+export const LibraryDataRequirementsArgs = CommonDataRequirementsArgs.extend({
+  expression: z.string()
+})
+  .partial()
+  .strict()
+  .superRefine(catchInvalidParams([catchMissingIdentifyingInfo], UNSUPPORTED_DATA_REQ_ARGS));
+
+export const CoreSearchArgs = z
+  .object({
+    'composed-of': z.string(),
+    context: z.string(),
+    'context-quantity': z.string(),
+    'context-type': z.string(),
+    'context-type-quantity': z.string(),
+    'context-type-value': z.string(),
+    date: checkDate('date'),
+    'depends-on': z.string(),
+    'derived-from': z.string(),
+    description: z.string(),
+    effective: z.string(),
+    identifier: z.string(),
+    jurisdiction: z.string(),
+    name: z.string(),
+    predecessor: z.string(),
+    status: z.string(),
+    successor: z.string(),
+    title: z.string(),
+    topic: z.string(),
+    url: z.string().url(),
+    version: z.string()
+  })
+  .partial()
+  .strict();
+
+export const LibrarySearchArgs = CoreSearchArgs.extend({
+  'content-type': z.string(),
+  type: z.string()
+})
+  .partial()
+  .strict()
+  .superRefine(catchInvalidParams([], UNSUPPORTED_LIBRARY_SEARCH_ARGS));
+
+export const MeasureSearchArgs = CoreSearchArgs.extend({
+  publisher: z.string()
```

Comment:
I see `publisher` listed for Library as well here: https://www.hl7.org/fhir/library.html#search
Should we move `publisher` to just appear under `CoreSearchArgs` instead?